tests/integration/steps/container_resources.go

```diff
@@ -0,0 +1,96 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/istio/operator/tests/integration/testcontext"
+	"github.com/pkg/errors"
+	"gopkg.in/inf.v0"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+	"strings"
+)
+
+type resourceStruct struct {
+	Cpu    resource.Quantity
+	Memory resource.Quantity
+}
+
+func DeploymentHasPodWithContainerResourcesSetToCpuAndMemory(ctx context.Context, depName, depNamespace, container, resourceType, cpu, memory string) error {
```

Comment:
Function name might be a little bit disguising since it doesn't really follow the deployment by OwnerReference by deployment name, and it uses the `app: label` selector instead, which might not always be present

Reply:
We use a similar approach when reading logs, so I thought about keeping it simple for the tests as well. But you are right, it might be confusing. Instead of iterating through OwnerReference (and adding a lot more lines), I think using the matchLabels gives the same level of confidence.